src/models/envelope/pae_v1.rs

```diff
@@ -0,0 +1,129 @@
+use std::str;
+
+use super::DSSEParser;
+use crate::Error;
+use crate::Result;
+
+const PREFIX: &str = "DSSEv1 ";
+const SPLIT: &str = " ";
+const SPLIT_U8: u8 = 0x20;
+
+pub struct PaeV1;
+
+/// Extract length and payload from bytes and consume them
+///
+/// length, SPLIT, next -> (length, next)
+fn consume_load_len(raw: &[u8]) -> Result<(usize, &[u8])> {
+    let mut iter = raw.splitn(2, |num| *num == SPLIT_U8);
+    let length_raw = iter.next().ok_or_else(|| {
+        Error::PAEParseFailed(format!("split '{}' failed for {:?}", SPLIT, raw.to_owned()))
+    })?;
+
+    let length = str::from_utf8(length_raw)?
+        .parse::<usize>()
+        .map_err(|_| Error::PAEParseFailed(format!("parse to int failed for {:?}", length_raw)))?;
+    let next = iter.next().ok_or_else(|| {
+        Error::PAEParseFailed(format!(
+            "prefix {} strip failed for {:?}",
+            PREFIX,
+            raw.to_owned()
+        ))
+    })?;
+    Ok((length, next))
+}
+
+impl DSSEParser for PaeV1 {
+    /// Use Pre-Authentication Encoding to pack payload for DSSE v1.
+    fn pae_pack(payload_ver: String, payload: &[u8]) -> Vec<u8> {
+        let sig_header: String = format!(
+            "{prefix}{payload_ver_len}{split}{payload_ver}{split}{payload_len}{split}",
```

Comment:
I recommend removing the trailing space and using a `split` after the prefix as well. This makes the separation very explicit and avoids trouble in the future with updates to the prefix since there's currently an unwritten expectation for it to end with a separator.